activities.py

```diff
@@ -1,150 +1,136 @@
+            return
+
+        # Validate required fields
         if 'issue' not in item:
-            self.log_error("Missing required 'issue' key", key, line)
+            self.log_error("Missing required 'issue' key.", key)
         elif not isinstance(item['issue'], int):
-            self.log_error(f"'issue' must be a unique number, got {item['issue']}", key, line)
-        elif item['issue'] in self.issues:
-            self.log_error(f"'issue' {item['issue']} is not unique", key, line)
-        else:
-            self.issues.add(item['issue'])
-
-        # Validate rationale (optional but must be string if present)
-        if 'rationale' in item and item['rationale'] is not None:
-            if not isinstance(item['rationale'], str):
-                self.log_error(f"'rationale' must be a string, got {item['rationale']}", key, line)
+            self.log_error(f"'issue' must be an integer, got {item['issue']}.", key)
+        elif item['issue'] < 0:
+            self.log_error(f"'issue' must be a positive integer, got {item['issue']}.", key)
 
         # Legacy item key restriction for issue numbers >= 1110
         if 'issue' in item and item['issue'] >= 1110:
             for legacy_key in ['position', 'venues']:
                 if legacy_key in item:
                     if self.fix:
-                        item.pop(legacy_key, None)
-                        modified = True
+                        del item[legacy_key]
                     else:
-                        self.log_error(f"Legacy key {legacy_key}. Please use GitHub issue labels instead.", key, line, fixable=True)
+                        self.log_error(f"Legacy key '{legacy_key}' is not allowed for issue numbers >= 1110.", key)
+
+        # Validate literal block fields
+        for field in ['description', 'rationale']:
+            self.validate_literal_block(data, key, field)
 
         # Optional fields validation
         if 'id' in item and (not isinstance(item['id'], str) or ' ' in item['id']):
-            self.log_error(f"'id' must be a string without whitespace, got {item['id']}", key, line)
-        if 'description' in item and item['description'] not in [None, '']:
-            if not isinstance(item['description'], str) or not item['description'].strip():
-                self.log_error(f"'description' must be a string or null, got {item['description']}", key, line)
+            self.log_error(f"'id' must be a string without whitespace, got {item['id']}.", key)
+
         if 'bug' in item and item['bug'] not in [None, '']:
             if not isinstance(item['bug'], str) or not item['bug'].startswith("https://bugzilla.mozilla.org/show_bug.cgi?id="):
-                self.log_error(f"'bug' must be a URL starting with 'https://bugzilla.mozilla.org/show_bug.cgi?id=', got {item['bug']}", key, line)
+                self.log_error(f"'bug' must be a URL starting with 'https://bugzilla.mozilla.org/show_bug.cgi?id=', got {item['bug']}.", key)
```

Comment:
If we're requiring this, why not include just the bug number (and take the `https://bugzilla...` as implied?)

Not for this change, but as a followup perhaps.

Reply:
I think it's easier for contributors to paste the full URL for all of the URLs than to find out what substring should be used. Though new issues will have the URLs in the issue OP instead, so it doesn't matter so much.